synapse/handlers/message.py

```diff
@@ -881,14 +892,6 @@ async def handle_new_client_event(
         else:
             room_version = await self.store.get_room_version_id(event.room_id)
 
-        event_allowed = await self.third_party_event_rules.check_event_allowed(
```

Comment:
this call path is somewhat less obvious than the others, since there are 6 callers of `handle_new_client_event`. They are:
 * `EventCreationHandler._send_dummy_event_for_room`
 * `EventCreationHandler.create_and_send_nonmember_event`
 * `RoomCreationHandler._upgrade_room`
 * `RoomMemberHandler._local_membership_update`
 * `RoomMemberHandler.send_membership_event`, which in turn is called by `FederationHandler.exchange_third_party_invite` and `FederationHandler.on_exchange_third_party_invite_request`
 * `RoomMemberMasterHandler._locally_reject_invite`

All the above apart from `RoomMemberMasterHandler._locally_reject_invite` create the event by calling `EventCreationHandler.create_event`, and thence `create_new_client_event`, and `RoomMemberMasterHandler._locally_reject_invite` is brought in line by #8537.